### ISSUE TITLE
[Bug]: GPU process launch failed when running from network share

### ISSUE THREAD

**Issue by mniesen** [electron/electron#36698]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

39.0.0

### What operating system are you using?

(tested on) Windows

### Operating System Version

(tested on) Windows 10 22H2 and Windows 11

### What arch are you using?

x64

### Last Known Working Electron version

13.6.9

### Expected Behavior

The GPU process launch error should not appear and the electron gui should open.

### Actual Behavior

When I run an electron program after version 13.6.9 from a mapped network share my electron app doesn't open and I get this error:
```
> electronversiontest@1.0.0 start
> electron .


[7896:0624/150429.768:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150429.910:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150429.919:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150429.925:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150429.960:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150429.969:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150430.051:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150430.080:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150430.082:ERROR:gpu_process_host.cc(996)] GPU process launch failed: error_code=18
[7896:0624/150430.082:FATAL:gpu_data_manager_impl_private.cc(449)] GPU process isn't usable. Goodbye.
```

The error does not appear if I run the exact same project from my local system drive (C:).

### Additional Information

The error does not appear when running electron with the --no-sandbox flag and the electron app opens.

**Comment by kianfobert** [comment]
It's failed launch for me please get it fixed i have tried multiple times to get it to work.

**Comment by mniesen** [comment]
Yes, this issue is still affecting me. This behavior is still happening in electron v23.1.4, v24.0.0-beta.3 and v25.0.0-nightly.20230317

**Comment by mniesen** [comment]
In the electron docs I can see that using --no-sandbox should be used for production.
<br>

![image](https://user-images.githubusercontent.com/50455466/228749326-6903ca07-7b04-4662-9e76-457185a9c91c.png)
(see: https://www.electronjs.org/docs/latest/tutorial/sandbox)

**Comment by t4551lo** [comment]
This affects us too. --no-sandbox is not an option for production environments. 

Currently we cannot build a portable client with a electron version above 13.6.9 but on the other hand some corporate firewalls start blocking clients with old versions of chromium (like in electron 13.6.9 which includes chromium [91.0.4472.164](https://source.chromium.org/chromium/chromium/src/+/refs/tags/91.0.4472.164:), released July 2021). 

So any improvement on this issue would be highly appreceated.

**Comment by mniesen** [comment]
To be clear, I can show on different devices completely separated with different versions of nodejs that there are problems from electron version 13.6.9. I find this very odd as I suspect that some companies using Electron use an SMB share so that the programmers' code isn't lost in addition to using a VCS.

**Comment by grozdanovgg** [comment]
bump

**Comment by techieanant** [comment]
bump

**Comment by grozdanovgg** [comment]
bump

**Comment by Withoutmakeup** [comment]
How to solve it

**Comment by mniesen** [comment]
> How to solve it

The solution still is to add a --no-sandbox flag when running the electron app, otherwise the error still appears.

**Comment by chs09** [comment]
Starting Electron with --in-process-gpu has provided us a solution that can be used productively without deactivating the sandbox functions.

**Comment by FabioHAraujo** [comment]
Same here

**Comment by Triangle4** [comment]
still persists with v31.3.1

**Comment by Thomas-QCD** [comment]
Created and ran the project on a network drive, which caused this error. I ran  `--in-process-gpu` using `app.commandLine.appendSwitch("--in-process-gpu");` at the top of my file, and the window opened but it was blank. I tried `--no-sandbox` and it worked fine, but there are some reasons not to deactivate the sandbox. I moved my project to my local drive, and it worked without any command line additions

**Comment by Ali-Bdeir** [comment]
bump

**Comment by chs09** [comment]
Unfortunately, the error still exists.
![Image](https://github.com/user-attachments/assets/ab06bc82-46b9-4947-8d8c-750e531f65e0)

Tested with the current Electron version 34.2.0

**Comment by yonson40** [comment]
help!!!

**Comment by mniesen** [comment]
Bump

**Comment by tiljes-st** [comment]
This error still persists. We're using redirection folders in our environment, so everything is technically on a network drive. What to do?

`--in-process-gpu` doesn't work. It will just show a blank screen. Electron 37.2.4.

**Comment by mniesen** [comment]
> This error still persists. We're using redirection folders in our environment, so everything is technically on a network drive. What to do?
> 
> `--in-process-gpu` doesn't work. It will just show a blank screen. Electron 37.2.4.

I think you mean folder redirection. If so, then we are both using a similar environment.

Indeed, this error still persists. After creating this issue here (2,5 years ago) I only found people that are experiencing the same problem. 

This problem is actually bigger than only Electron, because today no one makes programs anymore that work with UNC paths (except me and people who also use folder redirection). But folder redirection is indeed still used and should be accounted for by developers.

Hopefully this problem will be resolved one day (I hope sooner rather than later)

**Comment by tiljes-st** [comment]
Yes, folder redirection.

> This problem is actually bigger than only Electron

I dug around a bit more. It seems this is related to Chrome, which won't allow sandboxing on network locations. I didn't quite understanding the security implications behind it, though. Another reason might be that Windows by befault blocks (unsigned?) executables from network locations - because someone else could just swap out the exe with a malicious one, and the GPU access relies on some exe file.

However, if those reasons prove to be true, not much can be done. Except maybe trying to not make use of the gpu process at all?

**Comment by mniesen** [comment]
The security implications have to do with running chromium as multiple processes which would be more secure. But I too do not 
know what this means in practice.

What I do know is that I use folder redirection for roaming appdata, because log in and log out takes too long (+ 15 minutes) if you try to sync 10 Gb of roaming appdata. For file storage I'm using a home drive, this is a setting for my user in active directory. This implies that on log in a H:\ drive is mapped to folder on an SMB share. Even then something strange happens that is not path related, because C:\ and D:\ and V:\ work (SSD/HDD in my computer), but H:\ results in ```GPU process launch failed: error_code=18```.

So I believe it's something SMB related that chromium doesn't like.

**Comment by tiljes-st** [comment]
Yeah, it's not path related, but source related: even if mapped with drive letter or folder, Windows will still mark it as network share.

I think it is related to the security settings that would also cause this kind of dialog when running an executable from a network share:

<img width="459" height="285" alt="Image" src="https://github.com/user-attachments/assets/af7c4fc0-4b35-4971-a1d1-88a0220f38dd" />

**Comment by mniesen** [comment]
Is your PC domain join, because that notification should appear? It says that the file is from a locations outside of your local network.

It shouldn't be a red cross but an orange exclamation mark. 

<img width="466" height="359" alt="Image" src="https://github.com/user-attachments/assets/d8de5f5c-a882-4cee-9681-48425fba7086" />

I will do some research on this, if this could be the reason...

**Comment by tiljes-st** [comment]
Yes, it is domain joined. I just took the screenshot from the internet as an example.

Starting Electron Fiddle or other apps from my redirected folders doesn't cause a dialog though. Maybe because they have valid and trusted signatures?

**Comment by mniesen** [comment]
And is 192.168.13.10 domain joined?
Are you using Windows Security/Windows Defender or another kind of protection?

**Comment by tiljes-st** [comment]
> And is 192.168.13.10 domain joined?

This screenshot is just some random screenshot from the internet!

No Defender, we're using Trellix (new name of McAfee Enterprise solutions).

**Comment by mniesen** [comment]
bump
I still receive 
```
[31772:1029/120531.715:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.864:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.878:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.923:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.982:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.993:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.993:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120531.998:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120532.000:ERROR:content\browser\gpu\gpu_process_host.cc:993] GPU process launch failed: error_code=18
[31772:1029/120532.000:FATAL:content\browser\gpu\gpu_data_manager_impl_private.cc:415] GPU process isn't usable. Goodbye.
```
 in version 39.0.0

**Comment by shoustech** [comment]
Bump - currently on version v40.0.0

**Comment by tiljes-st** [comment]
From my understanding this is nothing that Electron can influence. It's in the design of Chromium.

**Comment by lwahonen** [comment]
Is there a way to mark this issue as not stale for the foreseeable future? I keep getting my hopes up because there was ticket activity, just to realize it was the bot again

**Comment by tiljes-st** [comment]
@lwahonen again, it's not an electron issue, but a Chromium issue. It should be closed and labeled "wontfix" and/or "upstream".

**Comment by mniesen** [comment]
I understand that this is due to a change in Chromium, but I can no longer write code with the Electron framework on a network drive. I hope that most developers do not write code stored locally on their devices, because if the SSD breaks or if a laptop is stolen, the uncommitted code is lost.

This doesn't effect Chromium because that will always be executed on the local devices.

**Comment by damianremington274-blip** [comment]
What version r u using exactly

**Comment by evanmorgan1231-dotcom** [comment]
I have questions about this part What version r u using exactly?

**Comment by mniesen** [comment]
bump bump bump
Still a problem, for over 3,5 years

**Comment by tiljes-st** [comment]
> bump bump bump Still a problem, for over 3,5 years

IT'S NOT AN ELECTRON ISSUE!!!!

It's a Chromium issue. File the bug report at Chromium please!

**Comment by mniesen** [comment]
> > bump bump bump Still a problem, for over 3,5 years
> 
> IT'S NOT AN ELECTRON ISSUE!!!!
> 
> It's a Chromium issue. File the bug report at Chromium please!

I understand the problem comes from chromium, but I'm not sure filing a bug report there will do anything.
Even if this is purely a chromium problem, it think it is only a problem in electron.
I don't think it is very weird for me to develop my program from a NAS, but it is weird to run chromium from a NAS, I think.

If it is a problem from chromium but electron includes this, than at the very least it should be explained that coding on a mapped SMB drive isn't supported.
Then I ask myself how everyone else is doing this. Is everyone really coding locally and using git often enough to have minimal lose in the even of a laptop being stolen or a disk dying?

**Comment by tiljes-st** [comment]
> > > bump bump bump Still a problem, for over 3,5 years
> > 
> > 
> > IT'S NOT AN ELECTRON ISSUE!!!!
> > It's a Chromium issue. File the bug report at Chromium please!
> 
> I understand the problem comes from chromium, but I'm not sure filing a bug report there will do anything.

Filing it here will do even less!

> Even if this is purely a chromium problem, it think it is only a problem in electron.

No. It affects EVERYTHING that is running in a sandboxed environment.

> I don't think it is very weird for me to develop my program from a NAS, but it is weird to run chromium from a NAS, I think.

I agree, but it's not possible with Chromium.

> If it is a problem from chromium but electron includes this, than at the very least it should be explained that coding on a mapped SMB drive isn't supported. Then I ask myself how everyone else is doing this. Is everyone really coding locally and using git often enough to have minimal lose in the even of a laptop being stolen or a disk dying?

But I don't think the issue tracker is the right place. It should be mentioned in the README/main page. And yes, I believe git (probably in connection with some IDE that makes pushing/pulling really easy). I mean, you can even have the git repo sitting on your local NAS.
